File: Ndapi/Block.cpp
#include "stdafx.h"
#include "Block.h"

namespace Ndapi
{
	Block::Block() : NdapiObject()
	{
	}

	Block::Block(d2fob* alert) : NdapiObject(alert)
	{
	}

	Block::Block(FormModule^ form, String^ name)
	{
		Create(name, form, D2FFO_BLOCK);
	}

	String^ Block::BackgroundColor::get()
	{
		return GetStringProperty(D2FP_BACK_COLOR);
	}

	void Block::BackgroundColor::set(String^ value)
	{
		SetStringProperty(D2FP_BACK_COLOR, value);
	}

	String^ Block::Description::get()
	{
		return GetStringProperty(D2FP_BLK_DESCRIPTION);
	}

	void Block::Description::set(String^ value)
	{
		SetStringProperty(D2FP_BLK_DESCRIPTION, value);
	}

	String^ Block::Comment::get()
	{
		return GetStringProperty(D2FP_COMMENT);
	}

	void Block::Comment::set(String^ value)
	{
		SetStringProperty(D2FP_COMMENT, value);
	}

	bool Block::DeleteAllowed::get()
	{
		return GetBooleanProperty(D2FP_DEL_ALLOWED);
	}

	void Block::DeleteAllowed::set(bool value)
	{
		SetBooleanProperty(D2FP_DEL_ALLOWED, value);
	}

	String^ Block::DeleteProcedureName::get()
	{
		return GetStringProperty(D2FP_DEL_PROC_NAM);
	}

	void Block::DeleteProcedureName::set(String^ value)
	{
		SetStringProperty(D2FP_DEL_PROC_NAM, value);
	}

	long Block::DMLArraySize::get()
	{
		return GetNumberProperty(D2FP_DML_ARY_SIZ);
	}

	void Block::DMLArraySize::set(long value)
	{
		SetNumberProperty(D2FP_DML_ARY_SIZ, value);
	}

	String^ Block::DMLDataTargetName::get()
	{
		return GetStringProperty(D2FP_DML_DAT_NAM);
	}

	void Block::DMLDataTargetName::set(String^ value)
	{
		SetStringProperty(D2FP_DML_DAT_NAM, value);
	}

	Ndapi::DMLDataTargetType Block::DMLDataTargetType::get()
	{
		return safe_cast<Ndapi::DMLDataTargetType>(GetNumberProperty(D2FP_DML_DAT_TYP));
	}

	void Block::DMLDataTargetType::set(Ndapi::DMLDataTargetType value)
	{
		SetNumberProperty(D2FP_DML_DAT_TYP, safe_cast<long>(value));
	}

	bool Block::EnforceColumnSecurity::get()
	{
		return GetBooleanProperty(D2FP_ENFRC_COL_SECURITY);
	}

	void Block::EnforceColumnSecurity::set(bool value)
	{
		SetBooleanProperty(D2FP_ENFRC_COL_SECURITY, value);
	}

	bool Block::EnforcePrimaryKey::get()
	{
		return GetBooleanProperty(D2FP_ENFRC_PRMRY_KEY);
	}

	void Block::EnforcePrimaryKey::set(bool value)
	{
		SetBooleanProperty(D2FP_ENFRC_PRMRY_KEY, value);
	}

	String^ Block::FillPattern::get()
	{
		return GetStringProperty(D2FP_FILL_PAT);
	}

	void Block::FillPattern::set(String^ value)
	{
		SetStringProperty(D2FP_FILL_PAT, value);
	}

	String^ Block::FontName::get()
	{
		return GetStringProperty(D2FP_FONT_NAM);
	}

	void Block::FontName::set(String^ value)
	{
		SetStringProperty(D2FP_FONT_NAM, value);
	}

	long Block::FontSize::get()
	{
		return GetNumberProperty(D2FP_FONT_SIZ);
	}

	void Block::FontSize::set(long value)
	{
		SetNumberProperty(D2FP_FONT_SIZ, value);
	}

	Ndapi::FontSpacing Block::FontSpacing::get()
	{
		return safe_cast<Ndapi::FontSpacing>(GetNumberProperty(D2FP_FONT_SPCING));
	}

	void Block::FontSpacing::set(Ndapi::FontSpacing value)
	{
		SetNumberProperty(D2FP_FONT_SPCING, safe_cast<long>(value));
	}

	Ndapi::FontStyle Block::FontStyle::get()
	{
		return safe_cast<Ndapi::FontStyle>(GetNumberProperty(D2FP_FONT_STY));
	}

	void Block::FontStyle::set(Ndapi::FontStyle value)
	{
		SetNumberProperty(D2FP_FONT_STY, safe_cast<long>(value));
	}

	Ndapi::FontWeight Block::FontWeight::get()
	{
		return safe_cast<Ndapi::FontWeight>(GetNumberProperty(D2FP_FONT_WGHT));
	}

	void Block::FontWeight::set(Ndapi::FontWeight value)
	{
		SetNumberProperty(D2FP_FONT_WGHT, safe_cast<long>(value));
	}

	String^ Block::ForegroundColor::get()
	{
		return GetStringProperty(D2FP_FORE_COLOR);
	}

	void Block::ForegroundColor::set(String^ value)
	{
		SetStringProperty(D2FP_FORE_COLOR, value);
	}

	bool Block::InsertAllowed::get()
	{
		return GetBooleanProperty(D2FP_INSRT_ALLOWED);
	}

	void Block::InsertAllowed::set(bool value)
	{
		SetBooleanProperty(D2FP_INSRT_ALLOWED, value);
	}

	String^ Block::InsertProcedureName::get()
	{
		return GetStringProperty(D2FP_INSRT_PROC_NAM);
	}

	void Block::InsertProcedureName::set(String^ value)
	{
		SetStringProperty(D2FP_INSRT_PROC_NAM, value);
	}

	Ndapi::KeyMode Block::KeyMode::get()
	{
		return safe_cast<Ndapi::KeyMode>(GetNumberProperty(D2FP_KEY_MODE));
	}

	void Block::KeyMode::set(Ndapi::KeyMode value)
	{
		SetNumberProperty(D2FP_KEY_MODE, safe_cast<long>(value));
	}

	Ndapi::LanguageDirection Block::LanguageDirection::get()
	{
		return safe_cast<Ndapi::LanguageDirection>(GetNumberProperty(D2FP_LANG_DIR));
	}

	void Block::LanguageDirection::set(Ndapi::LanguageDirection value)
	{
		SetNumberProperty(D2FP_LANG_DIR, safe_cast<long>(value));
	}

	Ndapi::LockingMode Block::LockingMode::get()
	{
		return safe_cast<Ndapi::LockingMode>(GetNumberProperty(D2FP_LOCK_MODE));
	}

	void Block::LockingMode::set(Ndapi::LockingMode value)
	{
		SetNumberProperty(D2FP_LOCK_MODE, safe_cast<long>(value));
	}

	String^ Block::LockProcedureName::get()
	{
		return GetStringProperty(D2FP_LOCK_PROC_NAM);
	}

	void Block::LockProcedureName::set(String^ value)
	{
		SetStringProperty(D2FP_LOCK_PROC_NAM, value);
	}

	bool Block::ListedInBlockMenu::get()
	{
		return GetBooleanProperty(D2FP_LST_IN_BLK_MNU);
	}

	void Block::ListedInBlockMenu::set(bool value)
	{
		SetBooleanProperty(D2FP_LST_IN_BLK_MNU, value);
	}

	long Block::MaximumQueryTime::get()
	{
		return GetNumberProperty(D2FP_MAX_QRY_TIME);
	}

	void Block::MaximumQueryTime::set(long value)
	{
		SetNumberProperty(D2FP_MAX_QRY_TIME, value);
	}

	long Block::MaxRecordsFetched::get()
	{
		return GetNumberProperty(D2FP_MAX_RECS_FETCHED);
	}

	void Block::MaxRecordsFetched::set(long value)
	{
		SetNumberProperty(D2FP_MAX_RECS_FETCHED, value);
	}

	String^ Block::Name::get()
	{
		return GetStringProperty(D2FP_NAME);
	}

	void Block::Name::set(String^ value)
	{
		SetStringProperty(D2FP_NAME, value);
	}

	Ndapi::NavigationStyle Block::NavigationStyle::get()
	{
		return safe_cast<Ndapi::NavigationStyle>(GetNumberProperty(D2FP_NAVIGATION_STY));
	}

	void Block::NavigationStyle::set(Ndapi::NavigationStyle value)
	{
		SetNumberProperty(D2FP_NAVIGATION_STY, safe_cast<long>(value));
	}

	String^ Block::NextNavigationBlock::get()
	{
		return GetStringProperty(D2FP_NXT_NAVIGATION_BLK_NAM);
	}

	void Block::NextNavigationBlock::set(String^ value)
	{
		SetStringProperty(D2FP_NXT_NAVIGATION_BLK_NAM, value);
	}

	String^ Block::OptimizerHint::get()
	{
		return GetStringProperty(D2FP_OPT_HINT);
	}

	void Block::OptimizerHint::set(String^ value)
	{
		SetStringProperty(D2FP_OPT_HINT, value);
	}

	String^ Block::PreviousNavigationBlock::get()
	{
		return GetStringProperty(D2FP_PREV_NAVIGATION_BLK_NAM);
	}

	void Block::PreviousNavigationBlock::set(String^ value)
	{
		SetStringProperty(D2FP_PREV_NAVIGATION_BLK_NAM, value);
	}

	bool Block::QueryAllowed::get()
	{
		return GetBooleanProperty(D2FP_QRY_ALLOWED);
	}

	void Block::QueryAllowed::set(bool value)
	{
		SetBooleanProperty(D2FP_QRY_ALLOWED, value);
	}

	bool Block::QueryAllRecord::get()
	{
		return GetBooleanProperty(D2FP_QRY_ALL_RECS);
	}

	void Block::QueryAllRecord::set(bool value)
	{
		SetBooleanProperty(D2FP_QRY_ALL_RECS, value);
	}

	String^ Block::QueryDataSourceName::get()
	{
		return GetStringProperty(D2FP_QRY_DAT_SRC_NAM);
	}

	void Block::QueryDataSourceName::set(String^ value)
	{
		SetStringProperty(D2FP_QRY_DAT_SRC_NAM, value);
	}

	long Block::QueryDataSourceType::get()
	{
		return GetNumberProperty(D2FP_QRY_DAT_SRC_TYP);
	}

	void Block::QueryDataSourceType::set(long value)
	{
		SetNumberProperty(D2FP_QRY_DAT_SRC_TYP, value);
	}

	long Block::NumberorRecordsBuffered::get()
	{
		return GetNumberProperty(D2FP_RECS_BUFFERED_COUNT);
	}

	void Block::NumberorRecordsBuffered::set(long value)
	{
		SetNumberProperty(D2FP_RECS_BUFFERED_COUNT, value);
	}

	long Block::NumberOfRecordsDisplayed::get()
	{
		return GetNumberProperty(D2FP_RECS_DISP_COUNT);
	}

	void Block::NumberOfRecordsDisplayed::set(long value)
	{
		SetNumberProperty(D2FP_RECS_DISP_COUNT, value);
	}

	long Block::RecordsToFetch::get()
	{
		return GetNumberProperty(D2FP_RECS_FETCHED_COUNT);
	}

	void Block::RecordsToFetch::set(long value)
	{
		SetNumberProperty(D2FP_RECS_FETCHED_COUNT, value);
	}

	Ndapi::RecordOrientation Block::RecordOrientation::get()
	{
		return safe_cast<Ndapi::RecordOrientation>(GetNumberProperty(D2FP_REC_ORNT));
	}

	void Block::RecordOrientation::set(Ndapi::RecordOrientation value)
	{
		SetNumberProperty(D2FP_REC_ORNT, safe_cast<long>(value));
	}

	String^ Block::CurrentRecordVisualAttributeGroup::get()
	{
		return GetStringProperty(D2FP_REC_VAT_GRP_NAM);
	}

	void Block::CurrentRecordVisualAttributeGroup::set(String^ value)
	{
		SetStringProperty(D2FP_REC_VAT_GRP_NAM, value);
	}

	NdapiObject^ Block::CurrentRecordVisualAttribute::get()
	{
		return GetObjectProperty<NdapiObject^>(D2FP_REC_VAT_GRP_OBJ);
	}

	void Block::CurrentRecordVisualAttribute::set(NdapiObject^ value)
	{
		SetObjectProperty(D2FP_REC_VAT_GRP_OBJ, value);
	}

	bool Block::ReverseDirection::get()
	{
		return GetBooleanProperty(D2FP_REV_DIR);
	}

	void Block::ReverseDirection::set(bool value)
	{
		SetBooleanProperty(D2FP_REV_DIR, value);
	}

	String^ Block::ScrollBarCanvas::get()
	{
		return GetStringProperty(D2FP_SCRLBR_CNV_NAM);
	}

	void Block::ScrollBarCanvas::set(String^ value)
	{
		SetStringProperty(D2FP_SCRLBR_CNV_NAM, value);
	}

	Ndapi::ScrollBarOrientation Block::ScrollBarOrientation::get()
	{
		return safe_cast<Ndapi::ScrollBarOrientation>(GetNumberProperty(D2FP_SCRLBR_ORNT));
	}

	void Block::ScrollBarOrientation::set(Ndapi::ScrollBarOrientation value)
	{
		SetNumberProperty(D2FP_SCRLBR_ORNT, safe_cast<long>(value));
	}

	String^ Block::ScrollBarTabPage::get()
	{
		return GetStringProperty(D2FP_SCRLBR_TBP_NAM);
	}

	void Block::ScrollBarTabPage::set(String^ value)
	{
		SetStringProperty(D2FP_SCRLBR_TBP_NAM, value);
	}

	long Block::ScrollBarWidth::get()
	{
		return GetNumberProperty(D2FP_SCRLBR_WID);
	}

	void Block::ScrollBarWidth::set(long value)
	{
		SetNumberProperty(D2FP_SCRLBR_WID, value);
	}

	long Block::ScrollBarXPosition::get()
	{
		return GetNumberProperty(D2FP_SCRLBR_X_POS);
	}

	void Block::ScrollBarXPosition::set(long value)
	{
		SetNumberProperty(D2FP_SCRLBR_X_POS, value);
	}

	long Block::ScrollBarYPosition::get()
	{
		return GetNumberProperty(D2FP_SCRLBR_Y_POS);
	}

	void Block::ScrollBarYPosition::set(long value)
	{
		SetNumberProperty(D2FP_SCRLBR_Y_POS, value);
	}

	bool Block::ShowScrollBar::get()
	{
		return GetBooleanProperty(D2FP_SHOW_SCRLBR);
	}

	void Block::ShowScrollBar::set(bool value)
	{
		SetBooleanProperty(D2FP_SHOW_SCRLBR, value);
	}

	bool Block::SingleRecord::get()
	{
		return GetBooleanProperty(D2FP_SNGL_REC);
	}

	void Block::SingleRecord::set(bool value)
	{
		SetBooleanProperty(D2FP_SNGL_REC, value);
	}

	bool Block::UpdateAllowed::get()
	{
		return GetBooleanProperty(D2FP_UPDT_ALLOWED);
	}

	void Block::UpdateAllowed::set(bool value)
	{
		SetBooleanProperty(D2FP_UPDT_ALLOWED, value);
	}

	bool Block::UpdateChangedColumnsOnly::get()
	{
		return GetBooleanProperty(D2FP_UPDT_CHANGED_COLS);
	}

	void Block::UpdateChangedColumnsOnly::set(bool value)
	{
		SetBooleanProperty(D2FP_UPDT_CHANGED_COLS, value);
	}

	String^ Block::UpdateProcedureName::get()
	{
		return GetStringProperty(D2FP_UPDT_PROC_NAM);
	}

	void Block::UpdateProcedureName::set(String^ value)
	{
		SetStringProperty(D2FP_UPDT_PROC_NAM, value);
	}

	String^ Block::VisualAttributeGroup::get()
	{
		return GetStringProperty(D2FP_VAT_NAM);
	}

	void Block::VisualAttributeGroup::set(String^ value)
	{
		SetStringProperty(D2FP_VAT_NAM, value);
	}

	String^ Block::WhereClause::get()
	{
		return GetStringProperty(D2FP_WHERE_CLAUSE);
	}

	void Block::WhereClause::set(String^ value)
	{
		SetStringProperty(D2FP_WHERE_CLAUSE, value);
	}

	bool Block::WhiteOnBlack::get()
	{
		return GetBooleanProperty(D2FP_WHITE_ON_BLACK);
	}

	void Block::WhiteOnBlack::set(bool value)
	{
		SetBooleanProperty(D2FP_WHITE_ON_BLACK, value);
	}

	NdapiEnumerator<BlockRelation^>^ Block::BlockRelations::get()
	{
		return gcnew NdapiEnumerator<BlockRelation^>(_handler, D2FP_REL);
	}

	NdapiEnumerator<Trigger^>^ Block::Triggers::get()
	{
		return gcnew NdapiEnumerator<Trigger^>(_handler, D2FP_TRIGGER);
	}
}

File: NdapiManaged/Block.cs
using NdapiManaged.Core.Handles;
using NdapiManaged.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace NdapiManaged
{
    public class Block : NdapiObject
    {
        public Block()
        {
        }

        public Block(string name)
        {
            Create(name, NdapiConstants.D2FFO_BLOCK);
        }

        internal Block(ObjectSafeHandle handle) : base(handle)
        {
        }

        public string BackgroundColor
        {
            get { return GetStringProperty(NdapiConstants.D2FP_BACK_COLOR); }
            set { SetStringProperty(NdapiConstants.D2FP_BACK_COLOR, value); }
        }

        public string Description
        {
            get { return GetStringProperty(NdapiConstants.D2FP_BLK_DESCRIPTION); }
            set { SetStringProperty(NdapiConstants.D2FP_BLK_DESCRIPTION, value); }
        }

        public string Comment
        {
            get { return GetStringProperty(NdapiConstants.D2FP_COMMENT); }
            set { SetStringProperty(NdapiConstants.D2FP_COMMENT, value); }
        }

        public bool DeleteAllowed
        {
            get { return GetBooleanProperty(NdapiConstants.D2FP_DEL_ALLOWED); }
            set { SetBooleanProperty(NdapiConstants.D2FP_DEL_ALLOWED, value); }
        }

        public string DeleteProcedureName
        {
            get { return GetStringProperty(NdapiConstants.D2FP_DEL_PROC_NAM); }
            set { SetStringProperty(NdapiConstants.D2FP_DEL_PROC_NAM, value); }
        }

        public int DMLArraySize
        {
            get { return GetNumberProperty(NdapiConstants.D2FP_DML_ARY_SIZ); }
            set { SetNumberProperty(NdapiConstants.D2FP_DML_ARY_SIZ, value); }
        }

        public string DMLDataTargetName
        {
            get { return GetStringProperty(NdapiConstants.D2FP_DML_DAT_NAM); }
            set { SetStringProperty(NdapiConstants.D2FP_DML_DAT_NAM, value); }
        }

        public DMLDataTargetType DMLDataTargetType
        {
            get { return GetNumberProperty<DMLDataTargetType>(NdapiConstants.D2FP_DML_DAT_TYP); }
            set { SetNumberProperty(NdapiConstants.D2FP_DML_DAT_TYP, value); }
        }

        public bool EnforceColumnSecurity
        {
            get { return GetBooleanProperty(NdapiConstants.D2FP_ENFRC_COL_SECURITY); }
            set { SetBooleanProperty(NdapiConstants.D2FP_ENFRC_COL_SECURITY, value); }
        }

        public bool EnforcePrimaryKey
        {
            get { return GetBooleanProperty(NdapiConstants.D2FP_ENFRC_PRMRY_KEY); }
            set { SetBooleanProperty(NdapiConstants.D2FP_ENFRC_PRMRY_KEY, value); }
        }

        public string FillPattern
        {
            get { return GetStringProperty(NdapiConstants.D2FP_FILL_PAT); }
            set { SetStringProperty(NdapiConstants.D2FP_FILL_PAT, value); }
        }

        public string FontName
        {
            get { return GetStringProperty(NdapiConstants.D2FP_FONT_NAM); }
            set { SetStringProperty(NdapiConstants.D2FP_FONT_NAM, value); }
        }

        public int FontSize
        {
            get { return GetNumberProperty(NdapiConstants.D2FP_FONT_SIZ); }
            set { SetNumberProperty(NdapiConstants.D2FP_FONT_SIZ, value); }
        }

        public FontSpacing FontSpacing
        {
            get { return GetNumberProperty<FontSpacing>(NdapiConstants.D2FP_FONT_SPCING); }
            set { SetNumberProperty(NdapiConstants.D2FP_FONT_SPCING, value); }
        }

        public FontStyle FontStyle
        {
            get { return GetNumberProperty<FontStyle>(NdapiConstants.D2FP_FONT_STY); }
            set { SetNumberProperty(NdapiConstants.D2FP_FONT_STY, value); }
        }

        public FontWeight FontWeight
        {
            get { return GetNumberProperty<FontWeight>(NdapiConstants.D2FP_FONT_WGHT); }
            set { SetNumberProperty(NdapiConstants.D2FP_FONT_WGHT, value); }
        }

        public string ForegroundColor
        {
            get { return GetStringProperty(NdapiConstants.D2FP_FORE_COLOR); }
            set { SetStringProperty(NdapiConstants.D2FP_FORE_COLOR, value); }
        }

        public bool InsertAllowed
        {
            get { return GetBooleanProperty(NdapiConstants.D2FP_INSRT_ALLOWED); }
            set { SetBooleanProperty(NdapiConstants.D2FP_INSRT_ALLOWED, value); }
        }

        public string InsertProcedureName
        {
            get { return GetStringProperty(NdapiConstants.D2FP_INSRT_PROC_NAM); }
            set { SetStringProperty(NdapiConstants.D2FP_INSRT_PROC_NAM, value); }
        }

        public KeyMode KeyMode
        {
            get { return GetNumberProperty<KeyMode>(NdapiConstants.D2FP_KEY_MODE); }
            set { SetNumberProperty(NdapiConstants.D2FP_KEY_MODE, value); }
        }

        public LanguageDirection LanguageDirection
        {
            get { return GetNumberProperty<LanguageDirection>(NdapiConstants.D2FP_LANG_DIR); }
            set { SetNumberProperty(NdapiConstants.D2FP_LANG_DIR, value); }
        }

        public LockingMode LockingMode
        {
            get { return GetNumberProperty<LockingMode>(NdapiConstants.D2FP_LOCK_MODE); }
            set { SetNumberProperty(NdapiConstants.D2FP_LOCK_MODE, value); }
        }

        public string LockProcedureName
        {
            get { return GetStringProperty(NdapiConstants.D2FP_LOCK_PROC_NAM); }
            set { SetStringProperty(NdapiConstants.D2FP_LOCK_PROC_NAM, value); }
        }

        public bool ListedInBlockMenu
        {
            get { return GetBooleanProperty(NdapiConstants.D2FP_LST_IN_BLK_MNU); }
            set { SetBooleanProperty(NdapiConstants.D2FP_LST_IN_BLK_MNU, value); }
        }

        public int MaximumQueryTime
        {
            get { return GetNumberProperty(NdapiConstants.D2FP_MAX_QRY_TIME); }
            set { SetNumberProperty(NdapiConstants.D2FP_MAX_QRY_TIME, value); }
        }

        public int MaxRecordsFetched
        {
            get { return GetNumberProperty(NdapiConstants.D2FP_MAX_RECS_FETCHED); }
            set { SetNumberProperty(NdapiConstants.D2FP_MAX_RECS_FETCHED, value); }
        }

        public NavigationStyle NavigationStyle
        {
            get { return GetNumberProperty<NavigationStyle>(NdapiConstants.D2FP_NAVIGATION_STY); }
            set { SetNumberProperty(NdapiConstants.D2FP_NAVIGATION_STY, value); }
        }

        public string NextNavigationBlock
        {
            get { return GetStringProperty(NdapiConstants.D2FP_NXT_NAVIGATION_BLK_NAM); }
            set { SetStringProperty(NdapiConstants.D2FP_NXT_NAVIGATION_BLK_NAM, value); }
        }

        public string OptimizerHint
        {
            get { return GetStringProperty(NdapiConstants.D2FP_OPT_HINT); }
            set { SetStringProperty(NdapiConstants.D2FP_OPT_HINT, value); }
        }

        public string PreviousNavigationBlock
        {
            get { return GetStringProperty(NdapiConstants.D2FP_PREV_NAVIGATION_BLK_NAM); }
            set { SetStringProperty(NdapiConstants.D2FP_PREV_NAVIGATION_BLK_NAM, value); }
        }

        public bool QueryAllowed
        {
            get { return GetBooleanProperty(NdapiConstants.D2FP_QRY_ALLOWED); }
            set { SetBooleanProperty(NdapiConstants.D2FP_QRY_ALLOWED, value); }
        }

        public bool QueryAllRecord
        {
            get { return GetBooleanProperty(NdapiConstants.D2FP_QRY_ALL_RECS); }
            set { SetBooleanProperty(NdapiConstants.D2FP_QRY_ALL_RECS, value); }
        }

        public string QueryDataSourceName
        {
            get { return GetStringProperty(NdapiConstants.D2FP_QRY_DAT_SRC_NAM); }
            set { SetStringProperty(NdapiConstants.D2FP_QRY_DAT_SRC_NAM, value); }
        }

        public int QueryDataSourceType
        {
            get { return GetNumberProperty(NdapiConstants.D2FP_QRY_DAT_SRC_TYP); }
            set { SetNumberProperty(NdapiConstants.D2FP_QRY_DAT_SRC_TYP, value); }
        }

        public int NumberorRecordsBuffered
        {
            get { return GetNumberProperty(NdapiConstants.D2FP_RECS_BUFFERED_COUNT); }
            set { SetNumberProperty(NdapiConstants.D2FP_RECS_BUFFERED_COUNT, value); }
        }

        public int NumberOfRecordsDisplayed
        {
            get { return GetNumberProperty(NdapiConstants.D2FP_RECS_DISP_COUNT); }
            set { SetNumberProperty(NdapiConstants.D2FP_RECS_DISP_COUNT, value); }
        }

        public int RecordsToFetch
        {
            get { return GetNumberProperty(NdapiConstants.D2FP_RECS_FETCHED_COUNT); }
            set { SetNumberProperty(NdapiConstants.D2FP_RECS_FETCHED_COUNT, value); }
        }

        public RecordOrientation RecordOrientation
        {
            get { return GetNumberProperty<RecordOrientation>(NdapiConstants.D2FP_REC_ORNT); }
            set { SetNumberProperty(NdapiConstants.D2FP_REC_ORNT, value); }
        }

        public string CurrentRecordVisualAttributeGroup
        {
            get { return GetStringProperty(NdapiConstants.D2FP_REC_VAT_GRP_NAM); }
            set { SetStringProperty(NdapiConstants.D2FP_REC_VAT_GRP_NAM, value); }
        }

        public NdapiObject CurrentRecordVisualAttribute
        {
            get { return GetObjectProperty<NdapiObject>(NdapiConstants.D2FP_REC_VAT_GRP_OBJ); }
            set { SetObjectProperty(NdapiConstants.D2FP_REC_VAT_GRP_OBJ, value); }
        }

        public bool ReverseDirection
        {
            get { return GetBooleanProperty(NdapiConstants.D2FP_REV_DIR); }
            set { SetBooleanProperty(NdapiConstants.D2FP_REV_DIR, value); }
        }

        public string ScrollBarCanvas
        {
            get { return GetStringProperty(NdapiConstants.D2FP_SCRLBR_CNV_NAM); }
            set { SetStringProperty(NdapiConstants.D2FP_SCRLBR_CNV_NAM, value); }
        }

        public ScrollBarOrientation ScrollBarOrientation
        {
            get { return GetNumberProperty<ScrollBarOrientation>(NdapiConstants.D2FP_SCRLBR_ORNT); }
            set { SetNumberProperty(NdapiConstants.D2FP_SCRLBR_ORNT, value); }
        }

        public string ScrollBarTabPage
        {
            get { return GetStringProperty(NdapiConstants.D2FP_SCRLBR_TBP_NAM); }
            set { SetStringProperty(NdapiConstants.D2FP_SCRLBR_TBP_NAM, value); }
        }

        public int ScrollBarWidth
        {
            get { return GetNumberProperty(NdapiConstants.D2FP_SCRLBR_WID); }
            set { SetNumberProperty(NdapiConstants.D2FP_SCRLBR_WID, value); }
        }

        public int ScrollBarXPosition
        {
            get { return GetNumberProperty(NdapiConstants.D2FP_SCRLBR_X_POS); }
            set { SetNumberProperty(NdapiConstants.D2FP_SCRLBR_X_POS, value); }
        }

        public int ScrollBarYPosition
        {
            get { return GetNumberProperty(NdapiConstants.D2FP_SCRLBR_Y_POS); }
            set { SetNumberProperty(NdapiConstants.D2FP_SCRLBR_Y_POS, value); }
        }

        public bool ShowScrollBar
        {
            get { return GetBooleanProperty(NdapiConstants.D2FP_SHOW_SCRLBR); }
            set { SetBooleanProperty(NdapiConstants.D2FP_SHOW_SCRLBR, value); }
        }

        public bool SingleRecord
        {
            get { return GetBooleanProperty(NdapiConstants.D2FP_SNGL_REC); }
            set { SetBooleanProperty(NdapiConstants.D2FP_SNGL_REC, value); }
        }

        public bool UpdateAllowed
        {
            get { return GetBooleanProperty(NdapiConstants.D2FP_UPDT_ALLOWED); }
            set { SetBooleanProperty(NdapiConstants.D2FP_UPDT_ALLOWED, value); }
        }

        public bool UpdateChangedColumnsOnly
        {
            get { return GetBooleanProperty(NdapiConstants.D2FP_UPDT_CHANGED_COLS); }
            set { SetBooleanProperty(NdapiConstants.D2FP_UPDT_CHANGED_COLS, value); }
        }

        public string UpdateProcedureName
        {
            get { return GetStringProperty(NdapiConstants.D2FP_UPDT_PROC_NAM); }
            set { SetStringProperty(NdapiConstants.D2FP_UPDT_PROC_NAM, value); }
        }

        public string VisualAttributeGroup
        {
            get { return GetStringProperty(NdapiConstants.D2FP_VAT_NAM); }
            set { SetStringProperty(NdapiConstants.D2FP_VAT_NAM, value); }
        }

        public string WhereClause
        {
            get { return GetStringProperty(NdapiConstants.D2FP_WHERE_CLAUSE); }
            set { SetStringProperty(NdapiConstants.D2FP_WHERE_CLAUSE, value); }
        }

        public bool WhiteOnBlack
        {
            get { return GetBooleanProperty(NdapiConstants.D2FP_WHITE_ON_BLACK); }
            set { SetBooleanProperty(NdapiConstants.D2FP_WHITE_ON_BLACK, value); }
        }

        public IEnumerable<Item> Items
        {
            get { return GetObjectList<Item>(NdapiConstants.D2FP_ITEM); }
        }

        public IEnumerable<BlockRelation> Relations
        {
            get { return GetObjectList<BlockRelation>(NdapiConstants.D2FP_REL); }
        }
        public IEnumerable<Trigger> Triggers
        {
            get { return GetObjectList<Trigger>(NdapiConstants.D2FP_TRIGGER); }
        }

    }
}
